ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/wizards/templates/PreprocessorParser.java

```diff
@@ -224,20 +214,13 @@ private void parse() throws Exception {
 
 			int opcode = 0;
 
-			switch (token) {
-				case T_AND :
-					opcode = OP_AND;
-					break;
-				case T_OR :
-					opcode = OP_OR;
-					break;
-				case T_EQ :
-					opcode = OP_EQ;
-					break;
-				case T_NEQ :
-					opcode = OP_NEQ;
-					break;
-			}
+			opcode = switch (token) {
+				case T_AND -> opcode = OP_AND;
+				case T_OR -> OP_OR;
+				case T_EQ -> OP_EQ;
+				case T_NEQ -> OP_NEQ;
+				default -> opcode;
```

Comment:
the default could be zero and the swtich expression could be assigned to the opcode variable during initialization already.